pkg/manifest/os_test.go

```diff
@@ -230,3 +230,21 @@ func testTomlPkgsFor(t *testing.T, os *OS) {
 		assert.Contains(t, buildPkgs, tc.expectedTomlPkg)
 	}
 }
+
+func TestFirstBootIncludesMachineIdStage(t *testing.T) {
```

Comment:
Thanks for the tests \o/

(nitpick) maybe a single test with a loop to test the two cases? but this if fine too of course
```go
func TestFirstBootIncludesMachineIdStage(t *testing.T) {
	os := NewTestOS()

	for _, firstBoot := range []bool{false, true} {
		os.FirstBoot = firstBoot
		pipeline := os.serialize()
		st := findStage("org.osbuild.machine-id", pipeline.Stages)
		if firstBoot {
			require.NotNil(t, st)
		} else {
			require.Nil(t, st)
		}
	}
}
````